### PR TITLE
feat: use timezone setting from user table

### PR DESCRIPTION
**Pull request by capJavert**
Remove `preferredTimezone` and always use timezone setting from `user` table.

### CODE REVIEW

File: src/cron/dailyDigest.ts [hunk 1]
```diff
@@ -35,17 +38,20 @@ const cron: Cron = {
...
       queryBuilder: personalizedDigestQuery,
       logger,
-      handler: async ({ personalizedDigest, emailBatchId }) => {
+      handler: async ({
+        personalizedDigest: personalizedDigestWithTimezome,
+        emailBatchId,
+      }) => {
+        const { timezone = 'Etc/UTC', ...personalizedDigest } =
+          personalizedDigestWithTimezome as UserPersonalizedDigest &
+            Pick<User, 'timezone'>;
```
Comment (capJavert) [inline src/cron/dailyDigest.ts:47]: Just to juggle type because typeorm types for `QueryBuilder` are garbage.

File: src/cron/dailyDigest.ts [hunk 2]
```diff
@@ -18,9 +19,11 @@ const cron: Cron = {
   handler: async (con, logger) => {
     const personalizedDigestQuery = con
       .createQueryBuilder()
+      .select('upd.*, u.timezone')
       .from(UserPersonalizedDigest, 'upd')
+      .leftJoin(User, 'u', 'u.id = upd."userId"')
       .where(
-        'clamp_to_hours("preferredHour" - EXTRACT(HOUR FROM NOW() AT TIME ZONE "preferredTimezone")) = :preferredHourOffset',
+        `clamp_to_hours("preferredHour" - EXTRACT(HOUR FROM NOW() AT TIME ZONE COALESCE(NULLIF(u.timezone, ''), 'Etc/UTC'))) = :preferredHourOffset`,
```
Comment (omBratteng) [inline src/cron/dailyDigest.ts:26]: Intended to change this to use backtick?

Comment (capJavert) [reply src/cron/dailyDigest.ts:26]: yes, due to `''` in query.

Comment (capJavert) [reply src/cron/dailyDigest.ts:26]: Yes I can also escape but this is more readable in my opinion.

File: src/cron/dailyDigest.ts [hunk 3]
```diff
@@ -35,17 +38,20 @@ const cron: Cron = {
     await schedulePersonalizedDigestSubscriptions({
       queryBuilder: personalizedDigestQuery,
       logger,
-      handler: async ({ personalizedDigest, emailBatchId }) => {
+      handler: async ({
+        personalizedDigest: personalizedDigestWithTimezome,
+        emailBatchId,
+      }) => {
+        const { timezone = 'Etc/UTC', ...personalizedDigest } =
```
Comment (idoshamun) [inline src/cron/dailyDigest.ts:45]: Can we store the default timezone somewhere if it's used multiple times?

**Review by idoshamun — APPROVED** [review]

File: src/cron/dailyDigest.ts [hunk 4]
```diff
@@ -18,9 +19,11 @@ const cron: Cron = {
   handler: async (con, logger) => {
     const personalizedDigestQuery = con
       .createQueryBuilder()
+      .select('upd.*, u.timezone')
       .from(UserPersonalizedDigest, 'upd')
+      .leftJoin(User, 'u', 'u.id = upd."userId"')
```
[the remark this answers is not in the document]
Comment (capJavert) [reply src/cron/dailyDigest.ts:24]: @idoshamun inner join?

[the remark this answers is not in the document]
Comment (idoshamun) [reply src/cron/dailyDigest.ts:24]: yes

File: src/cron/dailyDigest.ts (hunk 3 above)
Comment (capJavert) [reply src/cron/dailyDigest.ts]: Yes, added in 0c522af937cc06dec013abfed2363ff1fc9f16b7

File: src/cron/dailyDigest.ts (hunk 2 above)
Comment (omBratteng) [reply src/cron/dailyDigest.ts]: Ah, didn't catch the `''`, I was only looking for `${}`

File: src/cron/dailyDigest.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (capJavert) [reply src/cron/dailyDigest.ts:25]: Since tables have constraint on user id is it not the same?

[the remark this answers is not in the document]
Comment (idoshamun) [reply src/cron/dailyDigest.ts:25]: yes but `join` operation by default is inner join. using `leftJoin` means you have a reason to do that and you don't. so it's a cosmetic change

[the remark this answers is not in the document]
Comment (capJavert) [reply src/cron/dailyDigest.ts:25]: Makes sense 👍

**Pull request merged**